uberfire-widgets/uberfire-widgets-core/uberfire-widgets-core-client/src/main/java/org/uberfire/ext/widgets/core/client/tree/TreeItem.java

```diff
@@ -87,7 +88,6 @@ public TreeItem( final Type type,
                         public void onClick( ClickEvent event ) {
                             if ( !isSelected ) {
                                 updateSelected();
-                                return;
```

Comment:
@ederign Here solves the issue.. previously when the DOM element was clicked it first selected it and exited; now it selects it and opens it.